spec/lib/msf/ui/console/command_dispatcher/exploit_spec.rb

```diff
@@ -1,30 +1,343 @@
   end
 
-  describe "#cmd_run" do
+  describe '#cmd_check' do
+    context 'when running an exploit module' do
+      let(:current_mod) { exploit_mod }
+
+      it 'reports missing RHOST values' do
+        allow(current_mod).to receive(:run).and_call_original
+        current_mod.datastore['RHOSTS'] = ''
+        subject.cmd_check
+        expected_output = [
+          'Check failed: Msf::OptionValidateError One or more options failed to validate: RHOSTS.'
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+        expect(subject.mod).not_to have_received(:run)
+      end
+
+      it 'runs a single RHOST value' do
+        current_mod.datastore['RHOSTS'] = '192.0.2.1'
+        subject.cmd_check
+        expected_output = [
+          'Checking for target 192.0.2.1:3000 with normalized datastore value 3.5',
+          'Cleanup for target 192.0.2.1:3000',
+          '192.0.2.1:3000 - Check failed: The state could not be determined.'
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+      end
+
+      it 'runs multiple RHOST values' do
+        current_mod.datastore['RHOSTS'] = '192.0.2.1 192.0.2.2'
+        subject.cmd_check
+        expected_output = [
+          'Checking for target 192.0.2.1:3000 with normalized datastore value 3.5',
+          '192.0.2.1:3000 - Check failed: The state could not be determined.',
+          'Checking for target 192.0.2.2:3000 with normalized datastore value 3.5',
+          '192.0.2.2:3000 - Check failed: The state could not be determined.',
+          'Cleanup for target 192.0.2.1 192.0.2.2:3000'
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+      end
+
+      it 'normalizes the datastore before running' do
+        current_mod.datastore['RHOSTS'] = '192.0.2.1 192.0.2.2'
+        current_mod.datastore.store('FloatValue', '5.0')
+        subject.cmd_check
+        expected_output = [
+          'Checking for target 192.0.2.1:3000 with normalized datastore value 5.0',
+          '192.0.2.1:3000 - Check failed: The state could not be determined.',
+          'Checking for target 192.0.2.2:3000 with normalized datastore value 5.0',
+          '192.0.2.2:3000 - Check failed: The state could not be determined.',
+          'Cleanup for target 192.0.2.1 192.0.2.2:3000'
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+      end
+
+      it 'supports inline options' do
+        pending('cmd_check does not support inline values yet')
+        current_mod.datastore.store('FloatValue', '5.0')
+        subject.cmd_check('RHOSTS=192.0.2.5', 'FloatValue=10.0')
+        expected_output = [
+        ]
+
+        expect(@combined_output).to match_array(expected_output)
+      end
+
+      it 'supports multiple inlined RHOST values' do
+        pending('pending as inline module options are evaluated too late, and the module is therefore treated as a scanner')
```

Comment:
I've marked some tests as pending. I would expect this functionality to work as a user, but it doesn't work.